azurerm/resource_arm_kubernetes_cluster.go

```diff
@@ -107,6 +107,7 @@ func resourceArmKubernetesCluster() *schema.Resource {
 			"agent_pool_profile": {
 				Type:     schema.TypeList,
 				Required: true,
+				MaxItems: 1,
```

Comment:
@titilambert firstly sorry for the delay reviewing this!

I've spent the last few days working through trying to reconcile the open AKS PR's (#4676, #4543, #4472, #4256 and this one) to work out how to move forward.

Spending some time on this I believe there's a few things needed to fix this/get these in:

1. Introduce a new `default_node_pool` block which is limited to a single element, which replaces the existing `agent_pool_profile` block (which can then be removed in 2.0) which also allows us to resolve #4465
2. Supporting more conditional/delta updates within the AKS resource - allowing `ignore_changes` to properly work for the `count` field (amongst others) which fixes #4560, #4075, #3428 and #2993
3. Adding a more tests covering the open bugs/use-cases
4. Adding support for multiple node pools (via this PR) which fixes #3971 and #3835
5. Adding support for Standard Load Balancer via #4472

Whilst that's not ideal since 1-3 will supersede #4256 and #4543 (as the `agent_pool_profile` block is replaced) - this means users only have a breaking behavioural change once via the new block - unfortunately whatever happens it'll cause merge conflicts with #4472, but we can deal with that once the multiple node pools issue is resolved.

I've got the changes for 1-2 locally and I'm working through #3 at the moment - once that's done we should be able to get this merged :)

Since quite a few of those PR's are touching this file (`azurerm/resource_arm_kubernetes_cluster.go`) - I think it might be worth backing these changes out of this PR (via `git reset HEAD~<numberofcommits>`, rather than `git revert <sha>`) to avoid a series of merge conflicts with this PR - WDYT?

Reply:
I figured there might be some merge conflicts with my code. I'm also happy to create a new PR after this one gets merged, if that makes things easier. 👍 

Thanks for considering it.